### DISCUSSION TITLE
Does `"Pack All..."` menu option actually exist?

### DISCUSSION THREAD

**Discussion by kevinhartman** [NationalSecurityAgency/ghidra#4483] [Q&A]
I see in the Data Type Manager documentation that there should be a context menu option to pack all types in the current program into a project data type archive.

> Right-click on the program or data type archive where structures and unions are to be packed, and select the <I><B>Pack All...</B></I> action. A confirmation dialog will appear to make sure you want to pack all composites in the program or data type archive.

And here is [the quoted text in the docs](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/help/help/topics/DataTypeManagerPlugin/data_type_manager_description.htm#L483).

I cannot find a way to do this.

**Comment by Tim---** [comment]
Apparently, commit da800b6e41d47528fadfe33aa1e69fcc566ed5d8 removed the Pack* actions from the Data Type Manager:

https://github.com/NationalSecurityAgency/ghidra/blob/6fad151b5440de3dfdd619df7b5dc070b2b4dc6c/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/datamgr/DataTypesProvider.java#L157-L161

You can probably try to uncomment the lines for the actions that you need (or call them with a script ?).

**Reply by kevinhartman** [reply]
Huh. Thanks for the info.

I guess it wouldn't make sense in most cases to pack everything. Would be nice though if it was easier to say, pack a category in one go. Drag and drop is a pain.
